Lib/ipaddress.py

```diff
@@ -1546,7 +1546,7 @@ def __init__(self, address, strict=True):
         if self._prefixlen == (self.max_prefixlen - 1):
             self.hosts = self.__iter__
         elif self._prefixlen == (self.max_prefixlen):
-            self.hosts = lambda: [IPv4Address(addr)]
+            self.hosts = lambda: iter([IPv4Address(addr)])
```

Comment:
```suggestion
            self.hosts = lambda: iter((IPv4Address(addr),))
```

Very minor difference, but a quick `timeit.timeit()` suggests that constructing and using a tuple iterator is slightly faster on average (and I would imagine it uses slightly less memory).